relay-general/src/pii/processor.rs

```diff
@@ -59,6 +59,15 @@ impl<'a> Processor for PiiProcessor<'a> {
         meta: &mut Meta,
         state: &ProcessingState<'_>,
     ) -> ProcessingResult {
+        if let Some(crate::types::Value::String(original_value)) = meta.original_value_as_mut() {
+            if self
+                .apply_all_rules(None, state, Some(original_value))
+                .is_err()
```

Comment:
this implies the value should be hard-deleted. in that case it's safest to simply delete `original_value`.